frontend/src/views/c-summary.vue

```diff
@@ -487,25 +498,51 @@ export default defineComponent({
       // create deep clone of this.repos to not modify the original content of this.repos
       // when merging groups
       const groups = this.hasMergedGroups() ? JSON.parse(JSON.stringify(this.repos)) as Array<Repo> : this.repos;
-      groups.forEach((repo) => {
-        const res: Array<User> = [];
+      const res: Array<User> = [];
 
-        // filtering
-        repo.users?.forEach((user) => {
-          if (this.isMatchSearchedUser(this.filterSearch, user)) {
-            this.getUserCommits(user, this.filterSinceDate, this.filterUntilDate);
-            if (this.filterTimeFrame === 'week') {
-              this.splitCommitsWeek(user, this.filterSinceDate, this.filterUntilDate);
+      if (this.filterSearch.startsWith('tag:')) {
+        const tagSearch = this.filterSearch.split('tag:')[1];
+        groups.forEach((repo) => {
+          const commits = repo.commits;
+          if (!commits) return;
+
+          Object.entries(commits.authorDailyContributionsMap).forEach(([author, contributions]) => {
+            contributions = contributions as Array<AuthorDailyContributions>;
+            const tags = contributions.flatMap((c) => c.commitResults).flatMap((r) => r.tags);
+
+            if (tags.some((tag) => tag && this.isMatchSearchedTag(tagSearch, tag))) {
+              const user = repo.users?.find((u) => u.name === author);
+              if (user) {
+                this.updateCheckedFileTypeContribution(user);
+                res.push(user);
+              }
             }
-            this.updateCheckedFileTypeContribution(user);
-            res.push(user);
+          });
+
+          if (res.length) {
```

Comment:
This check is replicated in both the `if` and `else` clause. Is there a reason why we do it this way?

Reply:
no actually haha, let me change this